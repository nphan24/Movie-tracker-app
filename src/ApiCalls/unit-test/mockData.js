export const mockMovieData = {
  results: [
    {
      original_language: 'en',
      original_title: 'Jumanji: Welcome to the Jungle',
      overview:
        "The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
      popularity: 101.350698,
      poster_path: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
      release_date: '2017-12-09',
      title: 'Jumanji: Welcome to the Jungle',
      video: false,
      vote_average: 6.5,
      vote_count: 3115
    }
  ]
};

export const cleanedMovie = [
  {
    date: 'Release Date: 2017-12-09',
    id: 101.350698,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview:
      "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
    title: 'Jumanji: Welcome to the Jungle',
    vote: 'Rating: 6.5'
  }
];

export const movieInfo = {
  movie_id: '1',
  user_id: '24',
  title: 'hello',
  poster_path: '/hello',
  release_date: '04-01-18',
  vote_average: '-5',
  overview: 'Its not real'
};

export const mockUser = {
  id: '5',
  name: 'r',
  email: 'r@r.r',
  password: 'r'
};

export const mockFavorite = {
  date: 'Release Date: 2017-12-09',
  movie_id: 101.350698,
  image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
  overview:
    "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
  title: 'Jumanji: Welcome to the Jungle',
  vote: 'Rating: 6.5'
};

export const mockRemainingFavorite = {
  date: 'Release Date: 2018-10-5',
  movie_id: 10,
  image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
  overview: 'Summary: The tables are turned as four teenagers.',
  title: 'Fantastic Fours',
  vote: 'Rating: 2.5'
};

export const mockFavoritesArray = [
  {
    date: 'Release Date: 2018-10-5',
    movie_id: 10,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview: 'Summary: The tables are turned as four teenagers.',
    title: 'Fantastic Fours',
    vote: 'Rating: 2.5'
  },
  {
    date: 'Release Date: 2017-12-09',
    movie_id: 101.350698,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview:
      "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
    title: 'Jumanji: Welcome to the Jungle',
    vote: 'Rating: 6.5'
  }
];
