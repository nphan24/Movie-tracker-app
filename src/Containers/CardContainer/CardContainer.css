.card-container {
  background-color: #ECCC68;
  display: flex;
  justify-content: center;
}

.all-movies {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  min-height: 100vh;
}