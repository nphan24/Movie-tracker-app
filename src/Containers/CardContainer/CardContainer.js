import React from 'react';
import { Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import Card from '../../Components/Card/Card';
import PropTypes from 'prop-types';
import SingleMovie from '../../Components/SingleMovie/SingleMovie';

export const CardContainer = props => {
  const movieType =
    props.location.pathname === '/favorites' ? 'favorites' : 'movies';

  const renderCards = props[movieType].map(movie => {
    const cardClass = props.favorites.find(fav => fav.title === movie.title)
      ? 'selected'
      : '';

    return (
      <Card
        cardClass={cardClass}
        key={movie.title}
        movie={movie}
        movieType={movieType}
      />
    );
  });
  return (
    <section>
      <Route
        exact
        path="/movies/:id"
        render={({ match }) => {
          const movie = props.movies.find(
            movie => Number(match.params.id) === movie.movie_id
          );
          return <SingleMovie movie={movie} lastPath="" />;
        }}
      />
      <Route
        exact
        path="/"
        render={() => {
          return <div>{renderCards}</div>;
        }}
      />
      <Route
        exact
        path="/favorites"
        render={() => {
          return <div>{renderCards}</div>;
        }}
      />

      <Route
        exact
        path="/favorites/:id"
        render={({ match }) => {
          const movie = props.favorites.find(
            movie => Number(match.params.id) === movie.movie_id
          );
          return <SingleMovie movie={movie} lastPath="favorites" />;
        }}
      />
    </section>
  );
};

export const mapStateToProps = state => ({
  movies: state.movies,
  favorites: state.favorites,
  user: state.user
});

CardContainer.propTypes = {
  location: PropTypes.object,
  favorites: PropTypes.array,
  movies: PropTypes.array
};

export default withRouter(connect(mapStateToProps)(CardContainer));
