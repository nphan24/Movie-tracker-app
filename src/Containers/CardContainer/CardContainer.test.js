import { CardContainer, mapStateToProps } from './CardContainer';
import React from 'react';
import { shallow } from 'enzyme';
import * as mock from '../../ApiCalls/unit-test/mockData.js';
import { createLocation } from 'history';

describe('CardContainer', () => {
  let mockMovies;
  let mockFavs;
  beforeEach(() => {
    mockMovies = mock.cleanedMovie;
    mockFavs = mock.mockFavoritesArray;
  });
  it('should match the snapshot if on the home page', () => {
    const home = createLocation('/');
    const wrapper = shallow(
      <CardContainer location={home} movies={mockMovies} favorites={mockFavs}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot if on Favorites Page', () => {
    const favorites = createLocation('/favorites');
    const wrapper = shallow(
      <CardContainer location={favorites} favorites={mockMovies} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot on /favorites/:id', () => {
    const favSingle = createLocation('/favorites/4');
    const wrapper = shallow(<CardContainer location={favSingle} movies={mockMovies} favorites={mockFavs}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot on /movies/:id', () => {
    const movieSingle = createLocation('/movies/3');
    const wrapper = shallow(<CardContainer location={movieSingle} movies={mockMovies} favorites={mockFavs}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('correctly maps movies to props', () => {
    const mockMovies = mock.cleanedMovie
    const expected = mockMovies;
    const mockState = { movies: mockMovies };
    const mapped = mapStateToProps(mockState);
    expect(mapped.movies).toEqual(expected);
  });

  it('correctly maps favorites to props', () => {
    const mockFavs = mock.mockFavoritesArray;
    const expected = mockFavs;
    const mockState = {favorites: mockFavs};
    const mapped = mapStateToProps(mockState);
    expect(mapped.favorites).toEqual(expected);
  });

  it('correctly maps user to props', () => {
    const mockUser = mock.mockUser;
    const expected = mockUser;
    const mockState = {user: mockUser};
    const mapped = mapStateToProps(mockState);
    expect(mapped.user).toEqual(expected);
  }); 
});
