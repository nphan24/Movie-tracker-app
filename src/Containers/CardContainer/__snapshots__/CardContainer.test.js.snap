// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot if on Favorites Page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favorites={
        Array [
            Object {
              "movie_id": 215,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "poster_path": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "release_date": "Release Date: 2017-12-09",
              "title": "Jumanji: Welcome to the Jungle",
              "vote_average": "Rating: 6.5",
            },
          ]
    }
    location={
        Object {
            "hash": "",
            "pathname": "/favorites",
            "search": "",
            "state": undefined,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/movies/:id"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/favorites"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/favorites/:id"
          render={[Function]}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/movies/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/movies/:id"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/favorites/:id"
            render={[Function]}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/movies/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer should match the snapshot if on the home page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favorites={
        Array [
            Object {
              "date": "Release Date: 2018-10-5",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 10,
              "overview": "Summary: The tables are turned as four teenagers.",
              "title": "Fantastic Fours",
              "vote": "Rating: 2.5",
            },
            Object {
              "date": "Release Date: 2017-12-09",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 101.350698,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "title": "Jumanji: Welcome to the Jungle",
              "vote": "Rating: 6.5",
            },
          ]
    }
    location={
        Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
    }
    movies={
        Array [
            Object {
              "movie_id": 215,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "poster_path": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "release_date": "Release Date: 2017-12-09",
              "title": "Jumanji: Welcome to the Jungle",
              "vote_average": "Rating: 6.5",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/movies/:id"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/favorites"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/favorites/:id"
          render={[Function]}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/movies/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/movies/:id"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/favorites/:id"
            render={[Function]}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/movies/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer should match the snapshot on /favorites/:id 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favorites={
        Array [
            Object {
              "date": "Release Date: 2018-10-5",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 10,
              "overview": "Summary: The tables are turned as four teenagers.",
              "title": "Fantastic Fours",
              "vote": "Rating: 2.5",
            },
            Object {
              "date": "Release Date: 2017-12-09",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 101.350698,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "title": "Jumanji: Welcome to the Jungle",
              "vote": "Rating: 6.5",
            },
          ]
    }
    location={
        Object {
            "hash": "",
            "pathname": "/favorites/4",
            "search": "",
            "state": undefined,
          }
    }
    movies={
        Array [
            Object {
              "movie_id": 215,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "poster_path": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "release_date": "Release Date: 2017-12-09",
              "title": "Jumanji: Welcome to the Jungle",
              "vote_average": "Rating: 6.5",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/movies/:id"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/favorites"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/favorites/:id"
          render={[Function]}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/movies/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/movies/:id"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/favorites/:id"
            render={[Function]}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/movies/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer should match the snapshot on /movies/:id 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favorites={
        Array [
            Object {
              "date": "Release Date: 2018-10-5",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 10,
              "overview": "Summary: The tables are turned as four teenagers.",
              "title": "Fantastic Fours",
              "vote": "Rating: 2.5",
            },
            Object {
              "date": "Release Date: 2017-12-09",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 101.350698,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "title": "Jumanji: Welcome to the Jungle",
              "vote": "Rating: 6.5",
            },
          ]
    }
    location={
        Object {
            "hash": "",
            "pathname": "/movies/3",
            "search": "",
            "state": undefined,
          }
    }
    movies={
        Array [
            Object {
              "movie_id": 215,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "poster_path": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "release_date": "Release Date: 2017-12-09",
              "title": "Jumanji: Welcome to the Jungle",
              "vote_average": "Rating: 6.5",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/movies/:id"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/favorites"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/favorites/:id"
          render={[Function]}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/movies/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/movies/:id"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/favorites/:id"
            render={[Function]}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/movies/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
